core/src/context.rs

```diff
@@ -574,7 +574,17 @@ impl Context {
         if self.style.needs_relayout {
             apply_text_constraints(self, &tree);
 
-            apply_layout(&mut self.cache, &self.tree, &self.style);
+            // hack!
+            let mut store = (Style::default(), TextContext::default(), ResourceManager::default());
+            std::mem::swap(&mut store.0, &mut self.style);
+            std::mem::swap(&mut store.1, &mut self.text_context);
+            std::mem::swap(&mut store.2, &mut self.resource_manager);
+
+            apply_layout(&mut self.cache, &self.tree, &store);
+            std::mem::swap(&mut store.0, &mut self.style);
+            std::mem::swap(&mut store.1, &mut self.text_context);
+            std::mem::swap(&mut store.2, &mut self.resource_manager);
+
```

Comment:
This is what I was talking about when I said I hated the code. I couldn't get the borrow checker to be happy with pulling all those references out into a triple since there's the mutable borrow of cache, so I had to move ownership into the triple instead. If you can fix it, please PLEASE do.